File: src/types/prerequisites/single/StatePrerequisite.ts
import * as DB from "tsondb/schema/dsl"
import { StateIdentifier } from "../../_Identifier.js"
import { DisplayOption } from "../DisplayOption.js"

export const StatePrerequisite = DB.TypeAlias(import.meta.url, {
  name: "StatePrerequisite",
  comment: "Requires a specific state to be active.",
  type: () =>
    DB.Object({
      id: DB.Required({
        type: StateIdentifier(),
      }),
      display_option: DB.Optional({
        type: DB.IncludeIdentifier(DisplayOption),
      }),
    }),
})

File: Sources/OptolithDatabaseSchema/GeneratedTypes/prerequisites/single/StatePrerequisite.swift
//
//  StatePrerequisite.swift
//  OptolithDatabaseSchema
//

/// Requires a specific state or one of a specific set of states.
public struct StatePrerequisite: EntitySubtype {
    public let id: StateIdentifier

    public let displayOption: DisplayOption?

    public init(id: StateIdentifier, displayOption: DisplayOption? = nil) {
        self.id = id
        self.displayOption = displayOption
    }

    private enum CodingKeys: String, CodingKey {
        case id = "id"
        case displayOption = "display_option"
    }
}
